tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/BaseLinker.scala

```diff
@@ -388,19 +389,34 @@ final class BaseLinker(semantics: Semantics, esLevel: ESLevel, considerPositions
         OptimizerHints.empty, targetMDef.hash)
   }
 
-  private def findInheritedMethodDef(classInfo: Analysis.ClassInfo,
-      methodName: String, getTree: TreeProvider,
+  private def findInheritedMethodDef(analysis: Analysis,
+      classInfo: Analysis.ClassInfo, methodName: String, getTree: TreeProvider,
       p: Analysis.MethodInfo => Boolean = _ => true): MethodDef = {
     @tailrec
     def loop(ancestorInfo: Analysis.ClassInfo): MethodDef = {
       assert(ancestorInfo != null,
           s"Could not find $methodName anywhere in ${classInfo.encodedName}")
 
       val inherited = ancestorInfo.methodInfos.get(methodName)
-      if (inherited.exists(p)) {
-        findMethodDef(ancestorInfo, methodName, getTree)
-      } else {
-        loop(ancestorInfo.superClass)
+      inherited.find(p) match {
+        case Some(m) =>
+          m.syntheticKind match {
+            case MethodSyntheticKind.None =>
+              findMethodDef(ancestorInfo, methodName, getTree)
+
+            case MethodSyntheticKind.DefaultBridge(targetInterface) =>
+              val targetInterfaceInfo = analysis.classInfos(targetInterface)
+              findMethodDef(targetInterfaceInfo, methodName, getTree)
```

Comment:
Ah, that's what I got wrong. An the only reason we traverse through the superclasses is to reconstruct the argument types. Not for actual dispatch.

We should consider (in a future PR), to change the contents of the subclasses of `MethodSyntheticKind` to contain references to the actual `MethodInfos`, rather than just strings. This would not add any overhead in the `Analyzer`, but would simplify the `BaseLinker`.